tests/console/zypper_lr_validate.pm

```diff
@@ -15,6 +15,326 @@ use strict;
+        $product = 'SLE-Module-Web-Scripting';
+        $version = '12';
+    }
+    if ($product eq 'SLE-PHUB') {
+        $product = 'SUSE-PackageHub-';
+    }
+    # LTSS version is included in its product name
+    # leave it as empty to match the regex
+    if ($product =~ /LTSS/) {
+        $version = '';
+    }
+    diag "validatelr alias:$alias product:$product cha:$product_channel version:$version";
+
+    # Repo is checked for enabled/disabled state. If the information about the
+    # expected state is not delivered to validatelr(), we use some heuristics to
+    # determine the expected state: If the installation medium is a physical
+    # medium and the system is registered to SCC the repo should be disabled
+    # if the system is SLE 12 SP2 and later; enabled otherwise, see PR#11460 and
+    # FATE#320494.
+    my $scc_install_sle12sp2 = check_var('SCC_REGISTER', 'installation') and sle_version_at_least('12-SP2');
+    my $enabled_repo;
+    if ($args->{enabled_repo}) {
+        $enabled_repo = $args->{enabled_repo};
+    }
+    # bsc#1012258, bsc#793709: USB repo is disabled as the USB stick will be
+    # very likely removed from the system.
+    elsif ($args->{uri} =~ m{(cd|dvd|hd):///.*usb-}) {
+        $enabled_repo = 'No';
+    }
+    elsif ($args->{uri} =~ m{(cd|dvd|hd):///.*usbstick-}) {
+        record_soft_failure 'boo#1019634 repo on USB medium is not disabled for "hd:///…scsi…usbstick"';
+        $enabled_repo = 'Yes';
+    }
+    elsif ($args->{uri} =~ m{(cd|dvd|hd):///} and $scc_install_sle12sp2) {
+        $enabled_repo = 'No';
+    }
+    else {
+        $enabled_repo = 'Yes';
+    }
+    my $uri = $args->{uri};
+
+    my $cmd;
+    if ($product =~ /IBM-DLPAR-(Adv-Toolchain|SDK|utils)/) {
+        $cmd
+          = "zypper lr --uri | awk -F \'|\' -v OFS=\' \' \'{ print \$3,\$4,\$NF }\' | tr -s \' \' | grep --color \"$product\[\[:space:\]\[:punct:\]\[:space:\]\]*$enabled_repo $uri\"";
+    }
+    elsif (check_var('DISTRI', 'sle')) {
+        if (sle_version_at_least('15')) {
+            my $distri = uc(get_var('DISTRI'));
+            $cmd
+              = "zypper lr --uri | awk -F \'|\' -v OFS=\' \' \'{ print \$2,\$3,\$4,\$NF }\' | tr -s \' \' | grep --color \"$distri\[\[:alnum:\]\[:punct:\]\]*-*$version-$product_channel $distri\[\[:alnum:\]\[:punct:\]\[:space:\]\]*-*$version-$product_channel $enabled_repo $uri\"";
+        }
+        else {
+            # SLES12 does not have 'SLES12-Source-Pool' SCC channel
+            unless (($version eq "12") and ($product_channel eq "Source-Pool")) {
+                $cmd
+                  = "zypper lr --uri | awk -F \'|\' -v OFS=\' \' \'{ print \$2,\$3,\$4,\$NF }\' | tr -s \' \' | grep --color \"$product$version\[\[:alnum:\]\[:punct:\]\]*-*$product_channel $product$version\[\[:alnum:\]\[:punct:\]\[:space:\]\]*-*$product_channel $enabled_repo $uri\"";
+            }
+        }
+    }
+    run_scripted_command_slow($cmd, slow_type => 2) if defined $cmd;
```

Comment:
In all architectures we can't type this with normal speed ? 
Also if cmd is not defined , we just simply quit ? No error or at least warn message to a log ?
Suggesting to add option which will allow to use normal typing

Reply:
IIRC we have a ticket to get rid of "run_scripted_command_slow". I think it was not a good idea after all and should be removed -> handled in the ticket